### DISCUSSION TITLE
PollEvented in tokio 0.3

### DISCUSSION THREAD

**Discussion by yaa110** [tokio-rs/tokio#2964] [Q&A] [answered]
I have used `PollEvented` in [tokio-tun](https://github.com/yaa110/tokio-tun/blob/master/src/tun.rs#L17) to provide `AsyncRead` and `AsyncWrite`  implementations for a file descriptor. How can I replace it in my code?

```rust
pub struct TunIo(RawFd);

pub struct Tun {
    iface: Arc<Interface>,
    io: PollEvented<TunIo>,
}
```

**Comment by Darksonn** [comment, marked as answer]
This will be replaced by `AsyncFd` in #2903, which will hopefully be part of 0.3.1 soon.

**Reply by Darksonn** [reply]
Note that if you have an opinion on the destructor discussion in that PR, please do submit it.
